app/services/bill-runs/two-part-tariff/allocate-returns-to-charge-element.service.js

```diff
@@ -0,0 +1,100 @@
+'use strict'
+
+/**
+ * Allocates where applicable the abstracted volumes on the return log with the appropriate charge element
+ * @module AllocateReturnsToChargeElementService
+ */
+
+const { periodsOverlap } = require('../../../lib/general.lib.js')
+
+/**
+ * For a chargeElement with matching returns any abstracted volume recorded on the return log will be allocated to the
+ * charge element up to a maximum of the charge elements authorised volume, or the remaining authorised volume on the
+ * charge reference, whichever is lower.
+ *
+ * @param {module:ChargeElementModel} chargeElement - The charge element to match return logs against
+ * @param {module:ReturnLogModel[]} matchingReturns - logs that matched the charge element
+ * @param {module:ChargeVersionModel} chargePeriod - The charge period from the charge version the element belongs to
+ * @param {module:ChargeReferenceModel} chargeReference - The charge reference the element belongs to
```

Comment:
I know I wrote these comments but rereading them I think they could be better. Not sure what to change it to though 🫠